pilot/utils/style.py

```diff
@@ -1,46 +1,159 @@
+        'instruction': '#757575 bold'  # the color and style of the instruction - Grey
+    })
+
+    def __init__(self, theme):
+        """
+        Initializes a ThemeStyle instance.
+
+        Args:
+            theme (Theme): An enum member indicating the theme to use.
+        """
+        self.theme = theme
+
+    def get_style(self):
+        """
+        Returns the Style configuration for the current theme.
+
+        Returns:
+            questionary.Style: The Style instance for the current theme.
+        """
+        return self.DARK_STYLE if self.theme == Theme.DARK else self.LIGHT_STYLE
+
+
+class StyleConfig:
+    """
+    Class to manage the application's style and color configurations.
+    """
+    def __init__(self, theme: Theme = Theme.DARK):
+        """
+        Initializes a StyleConfig instance.
+
+        Args:
+            theme (Theme, optional): The initial theme to use. Defaults to Theme.DARK.
+        """
+        self.theme_style = ThemeStyle(theme)
+        self.theme = theme
+
+    def get_style(self):
+        """
+        Retrieves the Style configuration from the theme_style instance.
+
+        Returns:
+            questionary.Style: The Style configuration.
+        """
+        return self.theme_style.get_style()
+
+    def get_color(self, color_name: ColorName):
+        """
+        Retrieves the ANSI color code for the provided color_name, taking into account the current theme.
+
+        Args:
+            color_name (ColorName): Enum member indicating the desired color.
+
+        Returns:
+            str: The ANSI color code.
+        """
+        return color_name.value[self.theme == Theme.LIGHT]
+
+    def set_theme(self, theme: Theme):
+        """
+        Updates the theme of both the StyleConfig and its theme_style instance.
+
+        Args:
+            theme (Theme): Enum member indicating the new theme.
+        """
+        self.theme = theme
+        self.theme_style.theme = theme
+
+
+def get_color_function(color_name: ColorName, bold: bool = False):
+    """
+    Returns a function that colorizes text using the provided color_name and optionally makes it bold.
+
+    Args:
+        color_name (ColorName): Enum member indicating the color to use.
+        bold (bool, optional): If True, the returned function will bold text. Defaults to False.
+
+    Returns:
+        Callable[[str], str]: A function that takes a string and returns it colorized.
+    """
+    def color_func(text: str) -> str:
+        """
+        Colorizes the input text using the color and boldness provided when `get_color_function` was called.
+
+        Args:
+            text (str): The text to colorize.
+
+        Returns:
+            str: The colorized text.
+        """
+        color = style_config.get_color(color_name)
+        style = ColoramaStyle.BRIGHT if bold else ""
+        return f'{color}{style}{text}'

```

Comment:
Shouldn't we reset style after each message with Style.RESET_ALL? For example before we had:

```
def red(text):
    return f'{Fore.RED}{text}{Style.RESET_ALL}'
```

Reply:
#160 introduced init() for initialization. The purpose is to add the initialization of Colorama using colorama.init() at the beginning of the script.

```python
init(autoreset=True)
```
Here i use `autoreset=True`. By initializing with init(), the color auto-resets to default after every output.